cirq-core/cirq/work/sampler.py

```diff
@@ -311,32 +291,16 @@ async def run_batch_async(
         programs: Sequence['cirq.AbstractCircuit'],
         params_list: Optional[Sequence['cirq.Sweepable']] = None,
         repetitions: Union[int, Sequence[int]] = 1,
+        limiter: duet.Limiter = duet.Limiter(10),
     ) -> Sequence[Sequence['cirq.Result']]:
         """Runs the supplied circuits asynchronously.
 
         See docs for `cirq.Sampler.run_batch`.
         """
         params_list, repetitions = self._normalize_batch_args(programs, params_list, repetitions)
-        if len(programs) <= self.CHUNK_SIZE:
-            return await duet.pstarmap_async(
-                self.run_sweep_async, zip(programs, params_list, repetitions)
-            )
-
-        results = []
-        for program_chunk, params_chunk, reps_chunk in zip(
-            _chunked(programs, self.CHUNK_SIZE),
-            _chunked(params_list, self.CHUNK_SIZE),
-            _chunked(repetitions, self.CHUNK_SIZE),
-        ):
-            # Run_sweep_async for the current chunk
-            await duet.sleep(1)  # Delay for 1 second between chunk
-            results.extend(
-                await duet.pstarmap_async(
-                    self.run_sweep_async, zip(program_chunk, params_chunk, reps_chunk)
-                )
-            )
-
-        return results
+        return await duet.pstarmap_async(
+            self.run_sweep_async, zip(programs, params_list, repetitions, [limiter] * len(programs))
```

Comment:
shouldn't this be `repetitions), limit=...)`?

https://github.com/google/duet/blob/6817af592ecdf2aac0565a0de62bd7287fc9203a/duet/api.py#L171-L173

Reply:
That's for write requests... the intention is to pass a Limiter object to jobs.results_async so we can limit read requests from the spanner database